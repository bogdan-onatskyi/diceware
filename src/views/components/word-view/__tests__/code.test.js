import React from 'react';

import Code from '../code';

describe('Code component', () => {
    it('should render a code enabled', () => {
        const wrapper = shallow(
            <Code mod="1" word="word"/>
        );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.hasClass("wv__code--disabled")).toBe(false);
    });

    it('should render a code disabled', () => {
        const wrapper = shallow(
            <Code mod="1" word=""/>
        );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.hasClass("wv__code--disabled")).toBe(true);
    });

    it('should have 3 props', () => {
        const propsStr = 'className mod word';
        const props = [];
        for (let prop in Code.propTypes) props.push(prop);
        expect(props.join(' ') === propsStr).toBe(true);
    });
});
