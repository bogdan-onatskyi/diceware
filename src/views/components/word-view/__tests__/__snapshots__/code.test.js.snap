// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code component should render a code disabled 1`] = `
<div
  className="wv__code wv__code--1 wv__code--disabled"
/>
`;

exports[`Code component should render a code enabled 1`] = `
<div
  className="wv__code wv__code--1"
/>
`;
